frontend/src/scenes/experiments/experimentLogic.tsx

```diff
@@ -64,7 +72,37 @@ export const experimentLogic = kea<experimentLogicType>({
                     }
                     return { ...vals, ...experimentData }
                 },
-                emptyData: () => null,
+                updateExperimentGroup: (state, { variant, idx }) => {
+                    const featureFlagVariants = [...(state?.feature_flag_variants || [])]
+                    featureFlagVariants[idx] = { ...featureFlagVariants[idx], ...variant }
+                    return { ...state, feature_flag_variants: featureFlagVariants }
+                },
+                addExperimentGroup: (state) => {
```

Comment:
backend constraint: we have an upper limit of 4 total variants right now (or until I figure out how to calculate results for more variants :P )